File: 2024/JS/Day1/solutionA.js
const fs = require("fs");

fs.readFile("../../data/day1.txt", (err, data) => {
  const unorderedLeft = [];
  const unorderedRight = [];
  data
    .toString()
    .split("\n")
    .slice(0, -1)
    .forEach((row) => {
      const [l, r] = row.split("   ");
      unorderedLeft.push(l);
      unorderedRight.push(r);
    });

  const orderedLeft = unorderedLeft.sort((a, b) => a - b);
  const orderedRight = unorderedRight.sort((a, b) => a - b);

  const diff = orderedLeft.map((l, i) => Math.abs(l - orderedRight[i]));
  const sum = diff.reduce((acc, curr) => acc + curr, 0);

  console.log(sum);
});

File: 2024/Python/Day1/solutionA.py
left = []
right = []

with open("../../data/day1.txt") as file:
    for line in file:
        [l, r] = line.rstrip('\n').split("   ")
        left.append(int(l))
        right.append(int(r))

left.sort()
right.sort()

diffs = []

for i, l in enumerate(left):
    diffs.append(abs(right[i] - l))

print(sum(diffs))
